option.go

```diff
@@ -344,3 +346,13 @@ func (optSet *OptionSet) ByName(name string) *Option {
 	}
 	return nil
 }
+
+func (optSet *OptionSet) ByFlag(flag string) *Option {
+	for i := range *optSet {
+		opt := &(*optSet)[i]
```

Comment:
This doesn't work because we have a pointer to a slice. I assume `ByName` took an `*OptionSet` to be consistent with the other methods on it, namely `UnmarshalJSON` which requires the target to be a pointer (even for slices). I'll change the function signature to take in an `OptionSet` instead.